internal/controllers/helpers.go

```diff
@@ -285,3 +298,49 @@ func createObject(ctx context.Context, c client.Client, obj client.Object) error
 
 	return nil
 }
+
+func getTrustedCAcert(ctx context.Context, cl client.Client, agentTLSModeFeatureEnabled bool) ([]byte, error) {
+	log := log.FromContext(ctx)
+
+	if !agentTLSModeFeatureEnabled {
+		log.Info("agent-tls-mode feature is disabled, using system store")
+		return nil, nil
+	}
+
+	agentTLSModeSetting := &managementv3.Setting{}
+
+	if err := cl.Get(ctx, client.ObjectKey{
+		Name: "agent-tls-mode",
+	}, agentTLSModeSetting); err != nil {
+		if apierrors.IsNotFound(err) {
+			log.Info("agent-tls-mode setting not found, using system store")
+			return nil, nil // if setting is not found, return nil
+		}
+
+		return nil, fmt.Errorf("error getting agent-tls-mode setting: %w", err)
+	}
+
+	switch agentTLSModeSetting.Value {
+	case "system-store":
+		log.Info("using system store for CA certificates")
+		return nil, nil
+	case "strict":
+		log.Info("using strict mode for CA certificates")
+
+		caCertsSetting := &managementv3.Setting{}
+
+		if err := cl.Get(ctx, client.ObjectKey{
+			Name: "cacerts",
+		}, caCertsSetting); err != nil {
+			return nil, fmt.Errorf("error getting ca-certs setting: %w", err)
+		}
+
+		if caCertsSetting.Value == "" {
+			return nil, errors.New("ca-certs setting value is empty")
+		}
+
+		return []byte(caCertsSetting.Value), nil
+	default:
+		return nil, fmt.Errorf("invalid agent-tls-mode setting value: %s", agentTLSModeSetting.Value)
```

Comment:
It can actually be empty as well and treated as `system-store` setting: https://github.com/rancher/rancher/blob/649e7521bec6e4f2c83a8d3b9e4f1365b29e46e8/chart/values.schema.json#L6
```suggestion
		log.Info("using system store for CA certificates")
 		return nil, nil
```

Reply:
The setting should not be empty, if it's empty then we don't how to behave

Reply:
Does it mean the comment on the provided line is wrong? `agentTLSMode must be 'strict' or 'system-store' or null (defaults to system-store)`. It looks to me that `null` or empty is treated as `system-store`.

Reply:
Rancher UI doesn't allow an empty option, I assume it shouldn't be empty.

Reply:
The default should always be systemStore, not empty. It can be that users may nullify it manually through the Settings resource, but I would not support this scenario. 

For [reference](https://github.com/rancher/rancher/blob/de3ffa88cc75ae3da122bd36a4489663b5157ee3/pkg/settings/setting.go#L62):

```go
AgentTLSMode                        = NewSetting("agent-tls-mode", AgentTLSModeStrict).WithDefaultOnUpgrade(AgentTLSModeSystemStore)
```

Reply:
Ok, makes sense. If users left it empty, they’ll need to upgrade or fix it, with an error returned.